imagedephi/redact/redact.py

```diff
@@ -29,14 +31,11 @@ def get_base_rules():
 
 
 def iter_image_files(directory: Path) -> Generator[Path, None, None]:
-    """
-    Given a directory return an iterable of available images.
-
-    May raise a PermissionError if the directory is not readable.
-    """
+    """Given a directory return an iterable of available images."""
     for child in directory.iterdir():
-        if child.suffix in FILE_EXTENSION_MAP:
-            yield child
+        if child.is_file() and os.access(child, os.R_OK):
```

Comment:
[Given](https://docs.python.org/3/library/os.html#os.access):
> Note I/O operations may fail even when [access()](https://docs.python.org/3/library/os.html#os.access) indicates that they would succeed, particularly for operations on network filesystems which may have permissions semantics beyond the usual POSIX permission-bit model.

I'm worried that a very common source of permission-denied errors (users accessing files from a network share) will not be fully handled.